vms/evm/predicate/results.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package predicate
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/libevm/common"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/codec/linearcodec"
+	"github.com/ava-labs/avalanchego/utils/units"
+)
+
+const (
+	version        = uint16(0)
+	maxResultsSize = units.MiB
+)
+
+var resultsCodec codec.Manager
+
+func init() {
+	resultsCodec = codec.NewManager(maxResultsSize)
+
+	c := linearcodec.NewDefault()
+	err := resultsCodec.RegisterCodec(version, c)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// PrecompileResults is a map of results for each precompile address to the resulting byte array.
+type PrecompileResults map[common.Address][]byte
```

Comment:
I talked with Josh a bit in slack about this... I'm not sure we actually should do this, but we could have this be `map[common.Address]set.Bits`. The current `[]byte` is a bit more generic incase we ever added more precompiles with predicates... But currently we only use it for warp and these should (currently) always be `set.Bits`.

My gut reaction is that we should probably keep them as `[]byte`, but felt like it would make sense to write it down.

Reply:
I have changes it to `set.Bits` to move the bytes conversion away from the caller. I think this makes sense?